fix(discover): Supporting user field for top events

wmak opened the pull request:
- Fixes a bug when the user field is supplied for top events
- This widens the net quite a bit, as it does an IN for each user field
  and ORs them together.
- An alternative solution here is to skip user as a field to add to
  conditions, but I think that's worse since we'll get incorrect results
  in the final timeseries

wmak committed:
fix(discover): Supporting user field for top events

- Fixes a bug when the user field is supplied for top events
- This widens the net quite a bit, as it does an IN for each user field
  and ORs them together.
- An alternative solution here is to skip user as a field to add to
  conditions, but I think that's worse since we'll get incorrect results
  in the final timeseries

markstory commented on src/sentry/snuba/discover.py:
```diff
@@ -840,6 +840,14 @@ def top_events_timeseries(
             if field == "timestamp":
                 snuba_filter.conditions.append([["timestamp", "=", value] for value in values])
+            # A user field can be any of its field aliases, do an OR across all the user fields
+            elif field == "user":
```
Do we need special casing for `issue`?

wmak replied:
Yup, will make separate PR for that. I think that's the cause of top6 instead of top5 since the conditions aren't strict enough :+1:

evanh commented on tests/snuba/api/endpoints/test_organization_events_stats.py:
```diff
@@ -1024,3 +1024,32 @@ def test_top_events_with_int(self):
+        for key, result in six.iteritems(data):
+            # Because user represents all the possible user fields eg. user.id
+            # or user.email, the key is email followed by None several times
+            if key.startswith("bar@example.com"):
```
@markstory did some work to only return the email to the frontend for user when using the "user" alias. I think that might be beneficial here as well.

wmak committed:
fix: Use the same logic as transform to make user 1 field

- Also noticed that groupby had duplicates when there's user + user
  fields
  - eg. user + user.email results in user.email twice

wmak committed:
fix: can't use set since column_additions can be a list
